lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -351,8 +390,32 @@ static bool validateCodingKeysEnum(const DerivedConformance &derived,
     }
   }
 
-  if (!varDeclsAreValid)
+  if (!varDeclsAreValid) {
+    for (auto&& elt : extraKeys) {
+      Optional<Identifier> fixit = None;
+      unsigned bestScore = 0;
+      for (auto&& varDecl : varDecls) {
+        auto score = scoreParamAndArgNameTypo(elt->getBaseIdentifier().str(), varDecl.first.str(), bestScore);
+        if (score.hasValue() && (!fixit.hasValue() || score.getValue() < bestScore)) {
+          fixit = varDecl.first;
+          bestScore = score.getValue();
+        }
+      }
+      if (fixit.hasValue()) {
```

Comment:
As written, this algorithm can issue diagnostics for two different CodingKeys elements that are both similarly named for a parameter value. CSSimplify is working around this by striking paired parameters from consideration. https://github.com/apple/swift/blob/main/lib/Sema/CSSimplify.cpp#L670